lexicon/providers/godaddy.py

```diff
@@ -94,7 +94,7 @@ def _create_record(self, rtype, name, content):
         ttl = self._get_lexicon_option("ttl")
 
         # Retrieve existing data in DNS zone.
-        records = self._get("/domains/{0}/records".format(domain))
+        records = self._get("/domains/{0}/records/{1}/{2}".format(domain, rtype, relative_name))
```

Comment:
Given that the records are filtered on their type and relative_name now, I think that the condition on line 101-105 can be drastically simplified and test only for content equality.

Reply:
I agree that should be reasonable.  I left the original checks just to reduce the changes.  I've made this change as requested, too.